src/test/java/org/betonquest/betonquest/api/bukkit/config/custom/multi/MultiSectionConfigurationModificationTest.java

```diff
@@ -95,7 +154,23 @@ void testMergedResult() {
             yamlConfig.set(entry.getKey(), entry.getValue());
         }
 
-        assertEquals(configAll.saveToString(), yamlConfig.saveToString());
+        assertContainsAll(configAll, yamlConfig);
+        assertContainsAll(yamlConfig, configAll);
+    }
+
+    private void assertContainsAll(final ConfigurationSection config1, final ConfigurationSection config2) {
+        if (config1 == null) {
+            assertNull(config2);
+            return;
+        }
+        assertNotNull(config2);
+        for (final String key : config.getKeys(false)) {
+            if (config1.isConfigurationSection(key)) {
+                assertContainsAll(config1.getConfigurationSection(key), config2.getConfigurationSection(key));
+            } else {
+                assertEquals(config1.get(key), config2.get(key));
+            }
+        }
     }
```

Comment:
As this method is only used to check for equality (by calling it twice) it would be better to make it directly check for equality.

```suggestion
    private static void assertConfigurationEquals(final ConfigurationSection expected, final ConfigurationSection actual) {
        if (expected == null) {
            assertNull(actual);
            return;
        }
        assertNotNull(actual);

        final Set<String> expectedKeys = expected.getKeys(false);
        final Set<String> actualKeys = actual.getKeys(false);
        assertEquals(expectedKeys.size(), actualKeys.size(), "ConfigurationSections contain different amounts of keys.");
        assertTrue(expectedKeys.containsAll(actualKeys), "ConfigurationSections have differing keys.");

        for (final String key : expectedKeys) {
            if (expected.isConfigurationSection(key)) {
                assertTrue(actual.isConfigurationSection(key));
                assertConfigurationEquals(expected.getConfigurationSection(key), actual.getConfigurationSection(key));
            } else {
                assertEquals(expected.get(key), actual.get(key));
            }
        }
    }
```